distributed/actor.py

```diff
@@ -227,34 +276,73 @@ class ActorFuture:
     Actor
     """
 
-    def __init__(self, q, io_loop, result=None):
-        self.q = q
-        self.io_loop = io_loop
-        if result:
-            self._cached_result = result
-        self.status = "pending"
+    @abc.abstractmethod
+    def result(self, timeout: str | timedelta | float | None = None) -> _T:
+        ...
+
+    @abc.abstractmethod
+    def done(self) -> bool:
+        ...
+
+    def __repr__(self) -> Literal["<ActorFuture>"]:
+        return "<ActorFuture>"
+
 
-    def __await__(self):
+@dataclass(frozen=True, eq=False)
+class EagerActorFuture(BaseActorFuture[_T]):
+    """Future to an actor's method call when an actor calls another actor on the same worker"""
+
+    _result: _T
+
+    def __await__(self) -> Generator[object, None, _T]:
+        return self._result
+        yield
+
+    def result(self, timeout: object = None) -> _T:
+        return self._result
+
+    def done(self) -> Literal[True]:
+        return True
+
+
+@dataclass(frozen=True, eq=False)
+class _OK(Generic[_T]):
+    _v: _T
+
+    def unwrap(self) -> _T:
+        return self._v
+
+
+@dataclass(frozen=True, eq=False)
+class _Error:
+    _e: Exception
+
+    def unwrap(self) -> NoReturn:
+        raise self._e
+
+
+@dataclass(frozen=False, eq=False)
```

Comment:
Why make `ActorFuture` and `EagerActorFuture` dataclasses? I don't see much use of the dataclass sugar.

Reply:
Oh woops this line doesn't actually do anything useful here
```suggestion
```

Reply:
the usage on `EagerActorFuture` - with type annotations saves a fair bit of work, and `EagerActorFuture` especially make sense as a frozen dataclass as it just transports an already resolved value